File: Modulo16/Alunos/Aluno.cpp
#include "Aluno.h"

Aluno::Aluno() {
}

Aluno::Aluno(int RA, double NP, double NT) :
RA(RA), NP(NP), NT(NT) {
}

Aluno::~Aluno() {
}

int Aluno::getRA() const {
    return RA;
}

double Aluno::media() const {
    return NP * 0.8 + NT * 0.2;
}
void Aluno::imprime() {
    cout << "RA: " << RA;
    cout << ", NP: " << NP;
    cout << ", NT: " << NT;
    cout << ", Média: " << media() << endl;
}

bool Aluno::compara(const Aluno a1, const Aluno a2) {
    return a1.media() < a2.media();
}

File: Modulo06/AlunoEndereco-CSharp/Aluno.cs
using System;

public class Aluno {
    private string nome;
    private int RA;
    private Endereco endereco;

    public Aluno(string nome, int RA, Endereco endereco) {
        this.nome = nome;
        this.RA = RA;
        this.endereco = endereco;
    }

    public string getNome() {
        return this.nome;
    }

    public void setNome(string nome) {
        this.nome = nome;
    }

    public int getRA() {
        return this.RA;
    }

    public void setRA(int RA) {
        this.RA = RA;
    }

    public Endereco getEndereco() {
        return this.endereco;
    }

    public void setEndereco(Endereco endereco) {
        this.endereco = endereco;
    }

    public void imprime() {
        Console.WriteLine("-------------------------------------");
        Console.WriteLine("Nome: " + this.getNome());
        Console.WriteLine("RA: " + this.getRA());
        Console.WriteLine("<Endereco>");
        this.getEndereco().imprime();
    }
}